oxide/lib.go

```diff
@@ -118,6 +155,78 @@ func defaultUserAgent() string {
 	return fmt.Sprintf("oxide.go/%s", version)
 }
 
+func profileCredentials(cfg Config) (*hostCreds, error) {
+	configDir := cfg.ConfigDir
+	if configDir == "" {
+		homeDir, err := os.UserHomeDir()
+		if err != nil {
+			return nil, fmt.Errorf("unable to find user's home directory: %w", err)
+		}
+		configDir = filepath.Join(homeDir, ".config", "oxide")
+	}
+
+	profile := cfg.Profile
+
+	// Use explicitly configured profile over default when both are set.
+	if cfg.UseDefaultProfile && profile == "" {
+		var err error
+		profile, err = defaultProfile(filepath.Join(configDir, "config.toml"))
+		if err != nil {
+			return nil, fmt.Errorf("failed to get default profile: %w", err)
+		}
+	}
+
+	fileCreds, err := readCredentials(filepath.Join(configDir, "credentials.toml"), profile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get credentials for profile %q: %w", profile, err)
+	}
+
+	return fileCreds, nil
+}
+
+// defaultProfile returns the default profile from config.toml, if present.
+func defaultProfile(configPath string) (string, error) {
+	configFile, err := toml.LoadFile(configPath)
+	if err != nil {
+		return "", fmt.Errorf("failed to open config: %w", err)
+	}
+
+	if profileName := configFile.Get("default-profile"); profileName != nil {
+		return profileName.(string), nil
+	}
+
+	return "", errors.New("no default profile set")
+}
+
+// readCredentials returns the token and host associated with a profile.
+func readCredentials(credentialsPath, profileName string) (*hostCreds, error) {
```

Comment:
Nit:
```suggestion
func readProfileCredentials(credentialsPath, profileName string) (*hostCreds, error) {
```

Reply:
I agree this name was a bit vague, and also think `profileCredentials` and `readCredentials` were too similar.

Updated to `getProfile` and `parseCredentialsFile`, respectively.